src/tools/genpolicy/src/mount_and_storage.rs

```diff
@@ -173,6 +177,63 @@ fn get_empty_dir_mount_and_storage(
     });
 }
 
+fn get_persistent_volume_claim_mount(
+    settings: &settings::Settings,
+    yaml_mount: &pod::VolumeMount,
+    yaml_volume: &volume::Volume,
+    p_mounts: &mut Vec<policy::KataMount>,
+    storages: &mut Vec<agent::Storage>,
+    persistent_volume_claims: &Vec<PersistentVolumeClaim>,
+    propagation: &str,
+    access: &str,
+) {
+    let volume_pvc = yaml_volume.persistentVolumeClaim.as_ref().unwrap();
+    let is_blk_mount =
+        persistent_volume_claims.iter()
+        .find(|pvcResource| pvcResource.metadata.name.as_ref() == Some(&volume_pvc.claimName))
+        .and_then(|pvcResource| pvcResource.spec.storageClassName.as_ref())
+        .is_some_and(|sc| settings.common.virtio_blk_storage_classes.contains(sc));
+
+    if is_blk_mount {
+        let source = "$(spath)/$(b64-pci-device-id)".to_string();
+
+        storages.push(agent::Storage {
+            driver: "blk".to_string(),
+            driver_options: Vec::new(),
+            fs_group: None,
+            source: "$(pci-device-id)".to_string(),
+            mount_point: source.to_string(),
+            fstype: "ext4".to_string(), // TODO: Don't hardcode this.
```

Comment:
How hard is it to not hardcode this right now? Can we read this from the spec somehow?

Reply:
I think we should set this to `"$(fs-type)"` for now and not validate it (we're not validating this field for other storages). AFAIK extracting the FS type looks different for static vs. dynamic provisioning.

Reply:
Done above